proto/spacemesh/v1/smesher_types.proto

```diff
@@ -58,86 +57,74 @@ message CoinbaseResponse {
-    COMPUTE_API_CLASS_CPU = 1; // useful for testing on systems without a cuda or vulkan GPU
-    COMPUTE_API_CLASS_CUDA = 2;
-    COMPUTE_API_CLASS_VULKAN = 3;
+message PoSTSetupComputeProvidersRequest {
+    // Whether to run a short benchmarking session for each provider to evaluate its performance
+    bool benchmark = 1;
 }
 
-// Basic post data. Used for users to provide requested post data and by
-// other messages which contain post data.
-message PostData {
-    string path = 1; // User provided path to create the post data files at
-    uint64 data_size = 2; // Requested post data size
-    bool   append = 3; // Append to existing files if they exist. Otherwise overwrite.
-    bool   throttle = 4; // Throttle down setup phase computations while user is interactive on system
-    uint32 provider_id = 5; // A PostProvider id
+message PoSTSetupComputeProvidersResponse {
+    repeated PoSTSetupComputeProvider providers = 1;
 }
 
-message CreatePostDataRequest {
-    PostData data = 1;
+message PoSTSetupStatusResponse {
+    PoSTSetupStatus status = 1;
 }
 
-// Param passed to methods to indicate a request to delete data files
-message StopSmeshingRequest {
-    bool delete_files = 1;
+message PoSTSetupStatusStreamResponse {
+    PoSTSetupStatus status = 1;
 }
 
-message StopPostDataCreationSessionRequest {
-    bool delete_files = 1;
+message PoSTConfigResponse {
+    uint32 bits_per_label = 1;
+    uint64 labels_per_unit = 2;
+    uint32 min_num_units = 3;
+    uint32 max_num_units = 4;
 }
 
-// Proof of space data status
-message PostStatus {
-    PostData post_data = 1; // basic post data (set previously by the user)
-    enum FilesStatus {
-        FILES_STATUS_UNSPECIFIED = 0; // Lane's favorite impossible value
-        FILES_STATUS_NOT_FOUND = 1; // Expected data files do not exist
-        FILES_STATUS_PARTIAL = 2; // Some files exist and init can be continued (and may be in progress)
-        FILES_STATUS_COMPLETE = 3; // Expected data files are available and verified
-    }
-    FilesStatus files_status = 2;
-    bool init_in_progress = 3; // True if init process is currently in progress
-    uint64 bytes_written = 4; // Bytes written to the data files (actual total files sizes)
-    string error_message = 5; // Last init process error message
-    enum ErrorType {
-        ERROR_TYPE_UNSPECIFIED = 0; // Lane's favorite imposible value
-        ERROR_TYPE_FILE_NOT_FOUND = 1; // All expected post data files not found in expected path
-        ERROR_TYPE_READ_ERROR = 2; // Failure to read from a data file
-        ERROR_TYPE_WRITE_ERROR = 3; // Failure to write to a data file
+message PoSTSetupComputeProvider {
+    uint32 id = 1; // 0, 1, 2...
+    string model = 2; // e.g. Nvidia GTX 2700
+    enum ComputeApiClass {
+        COMPUTE_API_CLASS_UNSPECIFIED = 0;
+        COMPUTE_API_CLASS_CPU = 1; // useful for testing on systems without a cuda or vulkan GPU
+        COMPUTE_API_CLASS_CUDA = 2;
+        COMPUTE_API_CLASS_VULKAN = 3;
```

Comment:
Should we maybe add a `COMPUTE_API_CLASS_OTHER`?

Reply:
For what reason? These value correspond to the type of providers supported by the gpu-post lib. If/when we'll add new supported types of providers to gpu-post then we'll add them to the enum in future api releases. Don't see how other is useful here.